src/jit/emit.h

```diff
@@ -1577,7 +1577,18 @@ class emitter
 #endif // defined(_TARGET_X86_)
 
     size_t emitIssue1Instr(insGroup* ig, instrDesc* id, BYTE** dp);
+
+#ifdef _TARGET_XARCH_
+    template <bool generateCode = true>
     size_t emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp);
+    void emitPredictInstSize(instrDesc* id);
```

Comment:
I would describe this as doing more than predicting; it's actually setting the size. I think it might be better named as `emitUpdateInstSize` or `emitSetInstSize`